internal/transactionsubmission/transaction_worker.go

```diff
@@ -46,6 +39,7 @@ type TransactionWorker struct {
 	monitorSvc          tssMonitor.TSSMonitorService
 	eventProducer       events.Producer
 	jobUUID             string
+	handlerFactory      TransactionHandlerFactoryInterface
```

Comment:
I think you can store the `txHandler` itself as when the `NewTransactionWorker` function is called. That'd make it easier to be used across this file.